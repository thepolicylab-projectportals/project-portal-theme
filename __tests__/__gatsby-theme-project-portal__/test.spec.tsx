import React from "react"
import "@testing-library/jest-dom/extend-expect"
import { getPageQueryData } from "gatsby-plugin-testing"

import { render, screen } from "@testing-library/react"
import {
  ProjectStatus,
  HeaderWithImage,
  DevelopmentBanner,
} from "@thepolicylab-projectportals/gatsby-theme-project-portal/src/components"

// You have to write data-testid

// const Title = () => <h1 data-testid="hero-title">Gatsby is awesome!</h1>
// const ProjectStatusTest = () => (
//   <ProjectStatus data-testid="test-title" status={"open"} />
// )
// test("Displays the correct title", () => {
//   const { getByTestId } = render(<ProjectStatusTest />)

// Assertion

//expect(getByTestId("test-title")).toHaveTextContent("Gatsby is awesome!")
describe("Test if the ProjectStatus component is working as expected", () => {
  it("<ProjectStatus /> matches snapshot", () => {
    const component = render(
      <ProjectStatus data-testid="footer-copyright" status="open" />
    )
    expect(component.container).toMatchSnapshot()
  })
  // --> Test will pass
})

describe("test HeaderWithImage Component", () => {
  test("should render year if post is from 2019", () => {
    render(
      <HeaderWithImage
        lede="test lede"
        title="test title"
        imageSrc={"./__tests__/__testImages__/default-icon.png"}
      />
    )
    // const testImage = document.querySelector("img") as HTMLImageElement
    // expect(testImage.alt).toContain(
    //   "./__tests__/__testImages__/default-icon.png"
    // )
    expect(screen.getByText("test lede")).toBeTruthy()
    expect(screen.getByText("test title")).toBeTruthy()
  })
})

describe("test DevelopmentBanner Component", () => {
  test("should render year if post is from 2019", () => {
    const component = render(<DevelopmentBanner />)
    expect(component.container).toMatchSnapshot()
  })
})
