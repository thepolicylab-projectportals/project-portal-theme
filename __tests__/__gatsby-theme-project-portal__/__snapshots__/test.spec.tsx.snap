// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the ProjectStatus component is working as expected <ProjectStatus /> matches snapshot 1`] = `
<div>
  <div
    class="flex w-full"
  >
    <span
      class="pt-1 text-body text-primary align-inherit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </span>
    <span
      class="pl-2 uppercase"
    >
      Open
    </span>
  </div>
</div>
`;

exports[`test DevelopmentBanner Component should render year if post is from 2019 1`] = `
<div>
  <div
    class="py-6 overflow-hidden bg-warning text-center"
  >
    <h4>
      ⚠️ This is a beta site.
       
      <a
        class="underline hover:no-underline"
        href="/contact"
      >
        We welcome feedback!
      </a>
       
      ⚠️
    </h4>
  </div>
</div>
`;
